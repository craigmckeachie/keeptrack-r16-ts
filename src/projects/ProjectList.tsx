import React from 'react';
import { Project } from './Project';

interface ProjectListProps {
  projects: Project[];
}

class ProjectList extends React.Component<ProjectListProps> {
  render() {
    const { projects } = this.props;
    return <pre>{JSON.stringify(projects, null, ' ')}</pre>;
  }
}

export default ProjectList;
