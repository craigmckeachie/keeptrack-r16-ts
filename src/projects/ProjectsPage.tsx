import React, { Fragment } from 'react';
import { MOCK_PROJECTS } from './MockProjects';
import ProjectList from './ProjectList';

function ProjectsPage() {
  return (
    <Fragment>
      <h1>Projects</h1>
      <ProjectList projects={MOCK_PROJECTS} />
    </Fragment>
  );
}

export default ProjectsPage;
