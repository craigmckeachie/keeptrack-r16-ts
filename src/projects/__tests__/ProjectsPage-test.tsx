import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import { UnconnectedProjectsPage } from '../ProjectsPage';
import { MOCK_PROJECTS } from '../MockProjects';
import { ProjectListProps } from '../ProjectList';

describe('<ProjectsPage>', () => {
  let wrapper: ShallowWrapper;
  let onLoadMock = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <UnconnectedProjectsPage
        onLoad={onLoadMock}
        projects={MOCK_PROJECTS}
        page={1}
      />
    );
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('onLoad should be called with page number', () => {
    const pageNumber = 1;
    expect(onLoadMock).toBeCalledWith(pageNumber);
  });

  test('renders <ProjectList />', () => {
    let projectListWrapper = wrapper.find('ProjectList');
    expect(projectListWrapper.length).toBe(1);
    expect((projectListWrapper.props() as ProjectListProps).projects).toBe(
      MOCK_PROJECTS
    );
  });

  test('error displays', () => {
    wrapper.setProps({ error: 'Fail' });
    expect(wrapper.find('div.error').text()).toContain('Fail');
  });

  test('loading indicator displays', () => {
    wrapper.setProps({ loading: true });
    const spinnerWrapper = wrapper.find('span.spinner');
    expect(spinnerWrapper.exists()).toBeTruthy();
  });

  test('When clicking more records...onLoad should be called with next page number', () => {
    const moreButton = wrapper.findWhere(
      element => element.type() === 'button' && element.text() === 'More...'
    );
    expect(moreButton.exists()).toBeTruthy();
    moreButton.simulate('click');
    expect(onLoadMock).toBeCalledWith(2);
  });

  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
