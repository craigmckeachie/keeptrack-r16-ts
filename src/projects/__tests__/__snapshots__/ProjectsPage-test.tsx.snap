// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectsPage> snapshot 1`] = `ShallowWrapper {}`;
